### PR TITLE
Non-zero Exit on IOC's Discovered in Non-interactive Mode

### LINKED ISSUE #31: Non-zero Exit on IOC's Discovered in Non-interactive Mode
# 💡 Summary #

Use non-zero exit when IOC's are discovered in non-interactive mode to enhance automatic workflows.

## Motivation and context ##

Common tooling in automatic workflows involves using non-zero exit codes to enable decision making after process completion. Using additional exit codes would enhance CISA CHIRP's ability to be used within these contexts. 

## Implementation notes ##

CISA CHIRP would run plugins to completion and use reports to determine whether IOC's discovered is greater than 0. If any IOC's were discovered from the reports, we'd exit with a non-zero `sys.exit(1)` (see below).

Use the following exit codes for status indications (i.e. `sys.exit(number)`):
- 0 == successful completion, no IOC's detected
- 1 == successful completion, IOC's detected
- 2 == unsuccessful completion (errors, unexpectedly incomplete run, etc)

Avoiding specifics about IOC's detected in logs may be beneficial (as otherwise public- or near-public display of this information may be a vulnerability or liability). Propose using generic log message (or no log message at all, solely relying on exit code) to indicate IOC's were discovered but remove specific mention of which ones. Open to thoughts or suggestions here!

## Acceptance criteria ##

- [ ] CISA CHIRP runs all specified plugins through to report completion
- [ ] Successful run with no IOC's detected from reports emits exit code 0
- [ ] Successful run with IOC's detected from reports emits exit code 1
- [ ] Unsuccessful run (errors, unexpectedly incomplete, etc) emits exit code 2

### PR DESCRIPTION
**Pull request by d33bs**
# Non-zero Exit on IOC's Discovered in Non-interactive Mode #

## 🗣 Description ##

Seeks IOC detection count from run and exits with non-zero status in non-interactive mode, retaining existing functionality in interactive mode. Addresses #31 

## 💭 Motivation and context ##

Common tooling in automatic workflows involves using non-zero exit codes to enable decision making after process completion. Using additional exit codes enhances CISA CHIRP's ability to be used within these contexts.

## 🧪 Testing ##

Used pre-commit to ensure proper conformance with linting and style.

Used the following against the code related to this PR to test various scenarios:
```shell
# non-interactive mode with non-zero exit (CHIRP self-detects IOC's when targeting itself)
python chirp.py -p yara -t c:\\chirp\\** -o chirp_result -l debug --non-interactive

# non-interactive mode with zero exit (no detections)
python chirp.py -p yara -t c:\\no_ioc_dir\\** -o chirp_result -l debug --non-interactive

# interactive mode with zero exit after prompt
python chirp.py -p yara -t c:\\no_ioc_dir\\** -o chirp_result -l debug
```

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] _All_ future TODOs are captured in issues, which are referenced
      in code comments.
* [ ] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [] Tests have been added and/or modified to cover the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

**Comment by DeemOnSecurity** [comment]
@d33bs i will review this evening and merge the change with changes for v1.0.6

**Comment by d33bs** [comment]
@DeemOnSecurity Sounds good, thank you!

File: README.md [hunk 1]
```diff
@@ -115,6 +115,10 @@ python3 chirp.py -p registry yara -t c:\\target_dir\\** -o chirp_result -l debug
            [+] Done! Your results can be found at Z:\README\output.
 ```
 
+## Non-interactive Mode
```
Comment (DeemOnSecurity) [inline README.md]: This should be triple hashed, as we use the double hash for section headers of our README

File: chirp.py [hunk 2]
```diff
@@ -20,10 +20,18 @@
             70,
             "DONE! Your results can be found in {}".format(os.path.abspath(OUTPUT_DIR)),
         )
-        wait()
+        iocs_discovered = iocs_discovered()
+        # sleep to account for console interval
+        time.sleep(2)
```
Comment (DeemOnSecurity) [inline chirp.py]: This sleep should now be unnecessary with the native logging library in use.

File: chirp/common.py [hunk 3]
```diff
@@ -140,3 +143,18 @@ def wait() -> None:
...
+
+def iocs_discovered() -> bool:
+    """Determine whether iocs were discovered."""
+    report_files = glob.glob("{}/*".format(OUTPUT_DIR))
+    for report_file in report_files:
+        with open(report_file, "r") as f:
+            data = json.load(f)
+            if len(data) > 0:
+                logging.log(
+                    70, "Discovered IoC's, please see output reports for more details."
```
Comment (DeemOnSecurity) [inline chirp/common.py:158]: As we're using the `COMPLETED` code (70) for these messages, we probably want to change our default log level when non-interactive mode is invoked to only display 70+ logs, leaving the `-v` switch for verbose-ness if a user wants to run in non-interactive mode but wants full output.

**Review by DeemOnSecurity — CHANGES_REQUESTED** [review]
See my review comments

**Comment by d33bs** [comment]
Thank you for the comments @DeemOnSecurity ! Pushed another commit to address them.

**Review by DeemOnSecurity — APPROVED** [review]

**Pull request merged**